tests/unit/test_term_set.py

```diff
@@ -215,3 +217,125 @@ def test_wrapper_extend_error(self):
+        with self.assertRaises(ValueError):
+            tc.get_config('VectorData', 'hdmf-common11')
+
+    def test_get_config_container_error(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        with self.assertRaises(ValueError):
+            tc.get_config('VectorData11', 'hdmf-common')
+
+    def test_already_loaded_path_error(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        with self.assertRaises(ValueError):
+            tc.load_termset_config(config_path=path)
+
+    def test_load_two_unique_configs(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        path2 = 'tests/unit/hdmf_config2.yaml'
+        tc = TermSetConfigurator(path=path)
+        tc.load_termset_config(config_path=path2)
+        self.assertEqual(tc.path, [path, path2])
+
+
+class ExtensionContainer(Container):
+    __fields__ = ("description",)
+
+    def __init__(self, **kwargs):
+        description, namespace, type_map = popargs('description', 'namespace', 'type_map', kwargs)
+        super().__init__(**kwargs)
+        self.namespace = namespace
+        self.type_map = type_map
+        self.description = description
+
+
+class TestTermSetConfigVectorData(TestCase):
+    def setUp(self):
+        if not REQUIREMENTS_INSTALLED:
+            self.skipTest("optional LinkML module is not installed")
+
+        self.tm = load_termset_config(config_path='tests/unit/hdmf_config.yaml', return_map=True)
+
+    def tearDown(self):
+        unload_termset_config()
+
+    def test_load_config(self):
+        config = get_loaded_config()
+        self.assertEqual(config,
+        {'namespaces': {'hdmf-common': {'version': '3.12.2',
+        'data_types': {'VectorData': {'description': 'example_test_term_set.yaml'},
+                       'VectorIndex': {'data': '...'}}}}})
+
+    def test_validate_with_config(self):
+        data = VectorData(name='foo', data=[0], description='Homo sapiens')
+        self.assertEqual(data.description.value, 'Homo sapiens')
+
+    def test_namespace_warn(self):
+        with self.assertWarns(Warning):
+            ExtensionContainer(name='foo',
+                               namespace='foo',
+                               type_map=self.tm,
+                               description='Homo sapiens')
+
+    def test_container_type_warn(self):
+        with self.assertWarns(Warning):
+            ExtensionContainer(name='foo',
+                               namespace='hdmf-common',
+                               type_map=self.tm,
+                               description='Homo sapiens')
+
+    def test_already_wrapped_warn(self):
+        with self.assertWarns(Warning):
+            terms = TermSet(term_schema_path='tests/unit/example_test_term_set.yaml')
+            VectorData(name='foo',
+                       data=[0],
+                       description=TermSetWrapper(value='Homo sapiens', termset=terms))
```

Comment:
```suggestion
        terms = TermSet(term_schema_path='tests/unit/example_test_term_set.yaml')
        with self.assertWarns(Warning):
            VectorData(name='foo',
                       data=[0],
                       description=TermSetWrapper(value='Homo sapiens', termset=terms))
```
Wrap just the line that you expect the warning to come from